src/serac/numerics/functional/boundary_integral_kernels.hpp

```diff
@@ -303,24 +411,48 @@ void element_gradient_kernel(ExecArrayView<double, 3, ExecutionSpace::CPU> dK, d
   using test_element  = finite_element<g, test>;
   using trial_element = finite_element<g, trial>;
 
-  constexpr int nquad = num_quadrature_points(g, Q);
-
-  static constexpr TensorProductQuadratureRule<Q> rule{};
+  constexpr bool is_QOI = test::family == Family::QOI;
+  using padded_derivative_type [[maybe_unused]] =
+      std::conditional_t<is_QOI, tuple<derivatives_type, zero>, derivatives_type>;
+
+  RAJA::TypedRangeSegment<size_t>          elements_range(0, num_elements);
+  constexpr int                            nquad = num_quadrature_points(g, Q);
+  constexpr TensorProductQuadratureRule<Q> rule{};
+#ifdef USE_CUDA
+  using teams_e       = RAJA::LoopPolicy<RAJA::cuda_block_x_direct>;
+  using launch_policy = RAJA::LaunchPolicy<RAJA::cuda_launch_t<false>>;
+#else
+  using teams_e                    = RAJA::LoopPolicy<RAJA::seq_exec>;
+  using launch_policy              = RAJA::LaunchPolicy<RAJA::seq_launch_t>;
+#endif
+#ifdef USE_CUDA
```

Comment:
`USE_CUDA` is too generic but `SERAC_USE_CUDA` is set for the whole code when on blueos (maybe this needs to be rethought).

If we want to allow compiling Functional with and without cuda in the same build. Should this be `FUNCTIONAL_USE_CUDA`?